### DISCUSSION TITLE
How can I disable ContextMenu when mouse right click or long press?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#16784] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/a465d532-8d35-4913-b310-4dd943ee06c6)
I dislike the ContextMenu in TextBox while mouse right click, or long press in touch screen.

How can I disable it?

```
<Style Selector="TextBox”>
   <Setter Property="ContextMenu" Value="{x:Null}"></Setter>
</Style>
```

Even though I set the style, it doesn't work any. Why?

**Comment by stevemonaco** [comment, marked as answer]
You're setting the wrong property:

https://github.com/AvaloniaUI/Avalonia/blob/e627821ed3655fe0d26ade448dfc852443fb47fc/src/Avalonia.Themes.Simple/Controls/TextBox.xaml#L92

```xml
<Style Selector="TextBox">
    <Setter Property="ContextFlyout" Value="{x:Null}" />
</Style>
```

This works, at least for right-click on Windows desktop.

**Reply by kyuranger** [reply]
It works now! Thank you.

**Reply by maxkatz6** [reply]
Alternatively, you can handle ContextRequested event and set Handled = true on arguments.

**Reply by kyuranger** [reply]
Thanks all the same.
